### PR TITLE
Cleanup cluster-api-provider-openstack groups, promote jichenjc to admin

### PR DESCRIPTION
**Pull request by sbueringer**
Hey folks,

goal of this PR is to cleanup the CAPO admin and maintainer groups. 

According to a short Slack discussion (https://kubernetes.slack.com/archives/C8TSNPY4T/p1635322637149000) it should be fine to only include folks who are currently active maintainers of CAPO.

@chaosaffe @dims @luxas @roberthbailey @timothysc @chrigl Please let me know, in case it's not okay to remove you from those groups.

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold 
for feedback

/assign @hidekazuna @jichenjc 
for lgtm

/assign @vincepri @fabriziopandini 
for approve

In case anyone else should approve too, please just cc additional folks :)

**Comment by roberthbailey** [comment]
I have no issues being removed. In other places, I've seen the pattern to move folks to an "emeritus" block (which is ignored by prow) but in my case I was here for bootstrapping purposes and not because I was a significant contributor so I'm fine being removed.

**Comment by dims** [comment]
/approve
/lgtm

+1

**Comment by sbueringer** [comment]
> I have no issues being removed. In other places, I've seen the pattern to move folks to an "emeritus" block (which is ignored by prow) but in my case I was here for bootstrapping purposes and not because I was a significant contributor so I'm fine being removed.

Thx for the quick feedback and good point, but I think the `emeritus` block only exists in `OWNERS` files.

**Comment by roberthbailey** [comment]
My mistake - I wasn't reading carefully enough and saw "config/kubernetes-sigs/sig-cluster-lifecycle/OWNERS" and thought this was an owners file.

**Review by mdbooth — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
Thx everyone!
/hold cancel

**Pull request merged**
